src/main/java/soconnect/MainApp.java

```diff
@@ -106,11 +110,23 @@ private ReadOnlySoConnect initSoConnect(Storage storage) {
     private ReadOnlyTodoList initTodoList(Storage storage, ReadOnlySoConnect initialSoConnectData) {
         Optional<ReadOnlyTodoList> todoListOptional;
         try {
-            todoListOptional = storage.readTodoList(initialSoConnectData.getTagList());
+            todoListOptional = storage.readTodoList(initialSoConnectData);
             if (todoListOptional.isEmpty()) {
                 logger.info("Data file not found. Will be starting with a sample TodoList");
             }
-            return todoListOptional.orElseGet(SampleDataUtil::getSampleTodoList);
+            return todoListOptional.orElseGet(() ->
+            {
+                for (int i = 0; i < SampleDataUtil.getSampleTodos().length; i++) {
+                    Todo todo = SampleDataUtil.getSampleTodos()[i];
+                    List<Tag> todoTags = new ArrayList<>(todo.getTags());
+                    for (int j = 0; j < todoTags.size(); j++) {
+                        if (!initialSoConnectData.getTagList().contains(todoTags.get(j))) {
+                            initialSoConnectData.addTag(todoTags.get(j));
+                        }
+                    }
+                }
```

Comment:
Arrowhead code. Also, please take note of the LOC of the method when you are resolving this issue.